packages/react-hooks/src/useSetInterval.ts

```diff
@@ -0,0 +1,30 @@
+import { useConst } from './useConst';
+import { useEffect } from 'react';
+
+/**
+ * Returns a self disposing setTimeout function.
+ */
+export const useSetInterval = () => {
+  const intervalIds = useConst<number[]>([]);
+
+  useEffect(
+    () => () => {
+      for (const id of intervalIds) {
+        clearInterval(id);
+      }
+      intervalIds.splice(0);
+    },
+    [],
+  );
+
+  return (func: () => void, duration: number) => {
```

Comment:
you should `useConst` here so we don't create a new setInterval function on every render